examples/auth/auth_service.exs

```diff
@@ -0,0 +1,95 @@
+# Run this script from the Spell root with
+#
+#     mix spell.example.auth
+
+defmodule Auth do
+  @moduledoc """
+  The `RPC` module implements the example for asynchronous Remote Procedure Call.
+  """
+
+  defmodule Callee do
+    require Logger
+    defstruct [:callee, :procedure, timeout: 1000]
+
+    # Public Functions
+
+    @doc """
+    Start the callee with the passed procedure.
+    """
+    def start_link(topics, options \\ []) do
+      {:ok, spawn_link(fn -> init(topics, options) end)}
+    end
+
+    # Private Functions
+
+    defp init(procedure, options) do
+      {:ok, callee} = Auth.new_peer([Spell.Role.Callee], options)
+      {:ok, _register_id} = Spell.cast_register(callee, procedure)
+
+      %__MODULE__{callee: callee,
+                  procedure: procedure}
+        |> struct(options)
+        |> loop()
+    end
+
+    defp loop(state) do
+      receive do
+        :stop -> :ok
```

Comment:
This `:stop` message won't actually stop your process, since the `loop()` function will always be called, whatever message is received. Something like the following would only loop when it receives a message:

```elixir
defp loop(state) do
  receive do
    :stop -> :ok
    {Spell.Peer, pid, message} -> 
      handle_message(state, message)
      loop(state)
  after
    state.timeout -> {:error, :timeout}
  end
end
```

Note that this will probably result in your 1000ms timeout killing the process, since it no longer loops on timeout. Because the `Callee` is supposed to be waiting for auth calls, which might be quite sporadic, a default timeout of `:infinity`might be more appropriate.